glfs.c

```diff
@@ -263,6 +267,38 @@ static void gluster_cache_refresh(glfs_t *fs, const char *cfgstring)
 	}
 }
 
+static int gluster_cache_query_or_add(glfs_t **fs, gluster_server *entry,
+                                      char *config, bool *init)
+{
+	int ret = -1;
+
+	pthread_mutex_lock(&glfs_lock);
+
+	*fs = gluster_cache_query(entry, config);
+	if (*fs) {
+		*init = false;
+		ret = 0;
+		goto out;
+	}
+
+	*fs = glfs_new(entry->volname);
+	if (!*fs) {
+		tcmu_err("glfs_new failed: %m\n");
```

Comment:
That's a really valid point, thanks for the catch @seroyer, will make the changes and spin soon.
There is #160 for tcmu_dev_err changes, I want to include it there.